src/main_app/db/services/jobs_service.py

```
@@ -306,6 +307,24 @@ def cancel_job_db(job_id: int, job_type: str | None = None) -> bool:
+    """
+    Special case since it filters by multiple columns (id and job_type).
+    """
+    try:
+        affected_rows = (
+            db.session.query(JobRecord)
+            .filter(JobRecord.id == job_id, JobRecord.job_type == job_type)
+            .delete(synchronize_session=False)
+        )
+        db.session.commit()
+        return affected_rows > 0
+    except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)